lib/composites/menu/events/main.js

```diff
@@ -14,25 +14,34 @@ const ACTIVE_CLASS = 'dqpl-active';
 
 module.exports = (elements) => {
   /**
-   * Listen for clicks outside the menu (when
-   * its opened AND not locked) to close it
+   * Listen for clicks outside the sidebar and topBar
+   * and close them if open AND locked
    */
-  if (elements.menu && elements.trigger) {
+
+  if (elements.menu && elements.trigger && elements.topBar) {
     document.addEventListener('click', (e) => {
       const target = e.target;
       const isLocked = elements.menu.getAttribute('data-locked') === 'true';
-      const isWithin = closest(target, '.dqpl-side-bar');
+      const isWithinSideBar = closest(target, '.dqpl-side-bar');
+      const isWithinDropdown = closest(target, '.dqpl-top-bar li[data-dropdown-expanded]');
       const isHamburger = Classlist(target).contains('dqpl-menu-trigger') ||
         !!closest(target, '.dqpl-menu-trigger');
+      const expandedDropdownItem = document.querySelector('[data-dropdown-expanded="true"]');
+      const menuExpanded = elements.menu.getAttribute('aria-expanded') === 'true';
+
+      if (!isWithinDropdown && expandedDropdownItem) {
+        expandedDropdownItem.click();
+      }
 
-      if (isWithin || isHamburger || isLocked) {
+      if (isWithinSideBar || isHamburger || isLocked) {
```

Comment:
to reduce the amount of `if` blocks, and to make this easier to read, I'd suggest:
```js
if (!isWithin && !isHamburder && !isLocked && menuExpanded) {
  onTriggerClick();
}
```